Subject: [PATCH] cleanup python2
---
Index: submodules/c64ref/Source/c64disasm/txt2html.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/submodules/c64ref/Source/c64disasm/txt2html.py b/submodules/c64ref/Source/c64disasm/txt2html.py
--- a/submodules/c64ref/Source/c64disasm/txt2html.py	(revision 4274bd8782c5d3b18c68e6b9479b0ec751eb96b1)
+++ b/submodules/c64ref/Source/c64disasm/txt2html.py	(date 1680057291686)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
 #
 # txt2html.py - converts a "c64disasm" commented disassembly txt file
 # into cross-referenced HTML.
@@ -9,26 +9,26 @@
 import sys, re, cgi, os
 
 if len(sys.argv) < 2:
-	print"Usage: " + sys.argv[0] + " <filename>"
+	print("Usage: " + sys.argv[0] + " <filename>")
 	sys.exit(1)
 
 filename = sys.argv[1]
 
 lines = [line.rstrip() for line in open(filename)]
 
-print "<meta http-equiv=\"Content-type\" content=\"text/html; charset=utf-8\" />"
+print('<meta http-equiv="Content-type" content="text/html; charset=utf-8" />')
 
-print "<title>" + lines[0][2:] + "</title>"
-print "<h1>" + lines[0][2:] + "</h1>"
+print("<title>" + lines[0][2:] + "</title>")
+print("<h1>" + lines[0][2:] + "</h1>")
 
 f = os.popen("git log -1 --pretty=format:%h " + filename)
 revision = f.read()
-print "revision " + revision
+print("revision " + revision)
 
-print "<pre>"
+print("<pre>")
 
 previous_line = ''
-previous_line_was_heading = False;
+previous_line_was_heading = False
 cached_header = ''
 
 for line in lines[1:]:
@@ -37,13 +37,13 @@
 
 	if len(line) != 0 and line[0] == '-':
 		if len(line) > 1 and line[1] == '-':
-			print "<hr/>"
+			print("<hr/>")
 		else:
 			comment = cgi.escape(line[1:])
-			urls = re.findall('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', comment)
+			urls = re.findall('https?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*(),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', comment)
 			for url in urls:
 				comment = comment.replace(url, "<a href=\"" + url + "\">" + url + "</a>")
-			print comment
+			print(comment)
 		continue
 
 	left = line[:32]
@@ -71,18 +71,18 @@
 	
 	address = line[2:6]
 	if len(address) and address != '    ':
-		sys.stdout.write("<a name=\"" + address + "\"/>")
+		sys.stdout.write('<a name="' + address + '"/>')
 
 	if cached_header != '':
-		print "</pre><h2>" + cached_header + "</h2><pre>"
+		print("</pre><h2>" + cached_header + "</h2><pre>")
 		cached_header = ''
 
 	hex_numbers = re.findall(r'\$[0-9A-F][0-9A-F][0-9A-F][0-9A-F]', line)
 	for hex_number in hex_numbers:
 		if (hex_number[1] == 'A' or hex_number[1] == 'B' or hex_number[1] == 'E' or hex_number[1] == 'F'):
-			line = line.replace(hex_number, "<a href=\"#" + hex_number[1:] + "\">" + hex_number + "</a>")
-	print line
+			line = line.replace(hex_number, '<a href="#' + hex_number[1:] + '">' + hex_number + '</a>')
+	print(line)
 	previous_line = line
 
 
-print "</pre>"
+print("</pre>")
